src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -747,30 +747,40 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
                         };
 
                         if candidates.is_empty() {
-                            resolution_failure(cx, &item, path_str, &dox, link_range);
-                            // this could just be a normal link
-                            continue;
-                        }
-
-                        let len = candidates.clone().present_items().count();
-
-                        if len == 1 {
-                            candidates.present_items().next().unwrap()
-                        } else if len == 2 && is_derive_trait_collision(&candidates) {
-                            candidates.type_ns.unwrap()
+                            if path_str == "true" || path_str == "false" {
+                                match PRIMITIVES.iter().find(|(ty, _)| *ty == "bool") {
+                                    Some((_, res)) => (*res, Some("bool".to_owned())),
+                                    None => {
+                                        resolution_failure(cx, &item, path_str, &dox, link_range);
+                                        continue;
+                                    }
```

Comment:
Should this be always checking in a loop? Shouldn't we get `bool` only once or make this static?